src/xtgeo/surface/regular_surface.py

```diff
@@ -2541,21 +2558,21 @@ def slice_cube(
 
     def slice_cube_window(
         self,
-        cube,
-        zsurf=None,
-        other=None,
-        other_position="below",
-        sampling="nearest",
-        mask=True,
-        zrange=None,
-        ndiv=None,
-        attribute="max",
-        maskthreshold=0.1,
-        snapxy=False,
-        showprogress=False,
-        deadtraces=True,
-        algorithm=2,
-    ):
+        cube: xtgeo.Cube,
+        zsurf: Optional[xtgeo.RegularSurface] = None,
+        other: Optional[xtgeo.RegularSurface] = None,
+        other_position: str = "below",
+        sampling: Literal["nearest", "cube", "trilinear"] = "nearest",
+        mask: bool = True,
+        zrange: Optional[float] = None,
+        ndiv: Optional[int] = None,
+        attribute: Union[List[ValidAttrs], ValidAttrs] = "max",
+        maskthreshold: float = 0.1,
+        snapxy: bool = False,
+        showprogress: bool = False,
+        deadtraces: bool = True,
+        algorithm: Literal[1, 2, 3] = 2,
+    ) -> Tuple(None, Dict[xtgeo.RegularSurface]):
```

Comment:
Maybe I am just unfamiliar with some corner of typing, but what does this type indicate? A `(None, Dict[xtgeo.RegularSurface])` tuple, or an `Tuple[Optional[Dict[xtgeo.RegularSurface]]]`? In either case I think it should be the `[]` brackets

Reply:
It will return either None or a Dict, so I ended up with `Optional[Dict[xtgeo.RegularSurface]]`